ql/src/test/queries/clientpositive/cbo_groupby_remove_key.q

```diff
@@ -0,0 +1,18 @@
+CREATE TABLE passenger
+(
+    id       INT    NOT NULL,
+    fname    STRING NOT NULL,
+    lname    STRING NOT NULL,
+    passport STRING NOT NULL,
+    UNIQUE (id) DISABLE RELY,
+    UNIQUE (passport) DISABLE RELY,
+    UNIQUE (fname, lname) DISABLE RELY
+);
+
+EXPLAIN CBO SELECT id, COUNT(1) FROM passenger GROUP BY id, passport;
+EXPLAIN CBO SELECT passport, COUNT(1) FROM passenger GROUP BY id, passport;
+EXPLAIN CBO SELECT id, COUNT(1) FROM passenger GROUP BY id, fname, lname, passport;
+EXPLAIN CBO SELECT passport, COUNT(1) FROM passenger GROUP BY id, fname, lname, passport;
+EXPLAIN CBO SELECT fname, COUNT(1) FROM passenger GROUP BY id, fname, lname, passport;
+EXPLAIN CBO SELECT lname, COUNT(1) FROM passenger GROUP BY id, fname, lname, passport;
+EXPLAIN CBO SELECT fname, lname, COUNT(1) FROM passenger GROUP BY id, fname, lname, passport;
```

Comment:
In this case, is having group by fname,lname always a better plan? -- even if there is a different aggregate function?

Reply:
If we change the aggregate function the query changes as well. 

If we assume that the set of redundant columns remains the same after changing the aggregate function (e.g., `MAX(fname)`) then it's always a good idea to drop unnecessary processing so grouping on `fname, lname` is the best option. 

If the set of redundant columns changes (e.g., `MAX(passport)`) then this optimization will not trigger in the same way (`passport` column cannot be dropped).